docs/learn/streams.md

```diff
@@ -120,21 +178,106 @@ except OpenAIError as e:
+
+```python
+@openai.call(model="gpt-4o-mini", stream=True)
+@prompt_template("Recommend a {genre} book")
+def recommend_book(genre: str) -> openai.OpenAIStream:
+    ...
 
-Mirascope's `call` decorator provides proper type hints and safety when working with streams. When you enable streaming, the return type of your function will accurately reflect the stream type.
+stream = recommend_book("fantasy")
+# Your IDE will provide autocompletion for stream methods and properties
+```
 
-Your IDE will recognize the response object when streaming as a provider-specific `BaseStream` instance and provide appropriate autocompletion and type checking for its methods and properties, improving your development experience when working with streamed LLM responses.
+This type safety ensures that your IDE can provide appropriate autocompletion and type checking for the stream's methods and properties, improving your development experience when working with streamed LLM responses.
 
 ## Best Practices
 
-- Real-time Feedback: Use streaming for applications where users benefit from seeing results immediately, such as chatbots or writing assistants.
-- Progress Indicators: Implement progress bars or loading animations that update based on the streamed response, improving user experience for longer generations.
-- Incremental Processing: Process streamed content incrementally for large outputs, reducing memory usage and allowing for early termination if needed.
-- Timeout Handling: Implement timeouts for streamed responses to handle cases where the LLM might take too long to generate content.
+When working with streaming in Mirascope, consider the following best practices:
+
+1. **Real-time Feedback**: Use streaming for applications where users benefit from seeing results immediately, such as chatbots or writing assistants.
+
+   ```python
+   @openai.call(model="gpt-4o-mini", stream=True)
+   @prompt_template("Write a short story about {topic}")
+   def write_story(topic: str):
+       ...
+
+   print("Generating story...")
+   for chunk, _ in write_story("a magical forest"):
+       print(chunk.content, end="", flush=True)
+   # > Generating story...
+   # Once upon a time, in a magical forest...
+   ```
+
+2. **Progress Indicators**: Implement progress bars or loading animations that update based on the streamed response, improving user experience for longer generations.
+
+   ```python
+   from tqdm import tqdm
+
+   @openai.call(model="gpt-4o-mini", stream=True)
+   @prompt_template("Summarize the following text: {text}")
+   def summarize_text(text: str):
+       ...
+
+   text = "..." # Long text to summarize
+   with tqdm(total=100, desc="Summarizing") as pbar:
+       for i, (chunk, _) in enumerate(summarize_text(text)):
+           print(chunk.content, end="", flush=True)
```

Comment:
it seems weird to me that we're printing and updating the progress bar?

wouldn't it be better to stream, capture the text, update the progress bar, and then print the final amount when done?

also, how do we know the total is `100` here when we have no idea how many chunks there might be? honestly not sure how to implement a progress bar like this, might need a little more thought here